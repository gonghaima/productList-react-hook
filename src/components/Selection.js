import React from "react";
import { Store } from "../store";
import { fetchDataAction } from "../actions";

export default () => {
  const { state, dispatch } = React.useContext(Store);
  const { apiUrl, currentOffset } = state;
  return (
    <select
      onChange={e =>
        fetchDataAction(dispatch, apiUrl, currentOffset, e.target.value)
      }
    >
      {state.itemsPerPage.map(num => (
        <option value={num} key={`displayOption${num}`}>
          {num} per page
        </option>
      ))}
    </select>
  );
};
