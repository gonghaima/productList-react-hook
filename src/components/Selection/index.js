import React from "react";
import { withRouter } from "react-router-dom";
import { Store } from "../../store";
import { fetchDataAction } from "../../actions";

const Selection = routeProps => {
  const { state, dispatch } = React.useContext(Store);
  const { apiUrl, currentOffset, currentLimit } = state;
  return (
    <select
      onChange={e =>
        fetchDataAction(
          dispatch,
          apiUrl,
          currentOffset,
          e.target.value,
          routeProps.history
        )
      }
      value={currentLimit}
    >
      {state.itemsPerPage.map(num => (
        <option value={num} key={`displayOption${num}`}>
          {num} per page
        </option>
      ))}
    </select>
  );
};

export default withRouter(Selection);
