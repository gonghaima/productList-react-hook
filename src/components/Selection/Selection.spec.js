import React from "react";
import { shallow } from "enzyme";
import Selection from "./index";

it("Selection component renders without crashing", () => {
  const SelectionComponent = shallow(<Selection />);
  expect(SelectionComponent.debug()).toMatchSnapshot();
});
