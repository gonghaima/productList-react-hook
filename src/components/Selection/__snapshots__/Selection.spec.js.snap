// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection component renders without crashing 1`] = `
"<ContextConsumer>
  [function]
</ContextConsumer>"
`;
